contributors/guide/first-contribution.md

```diff
@@ -34,6 +34,14 @@ If the above suggestions don't appeal to you, you can browse the
 in contributing without writing code can also find ideas in the 
 [Non-Code Contributions Guide].
 
+{{< note >}}
+Although contributions are welcome, beware that every pull
+request creates work for maintainers and costs for testing it.
+Fixing *linter warnings* is often *not* worth it because the
+existing code is fine. Always discuss with maintainers first
+before creating such PRs.
```

Comment:
If it turns out that fixes for certain types of linter warnings are always approved, this can be expanded... (One thing I have in mind is dealing with deprecations where the fix allows a dependency to be dropped, or a dependency migration to be completed.)